File: Kyoo/Models/Season.cs
using Newtonsoft.Json;

namespace Kyoo.Models
{
    public class Season
    {
        [JsonIgnore] public readonly long id;
        [JsonIgnore] public long ShowID;

        public long seasonNumber;
        public string Title;
        public string Overview;
        public long? year;

        [JsonIgnore] public string ImgPrimary;
        public string ExternalIDs;

        public Season() { }

        public Season(long id, long showID, long seasonNumber, string title, string overview, long? year, string imgPrimary, string externalIDs)
        {
            this.id = id;
            ShowID = showID;
            this.seasonNumber = seasonNumber;
            Title = title;
            Overview = overview;
            this.year = year;
            ImgPrimary = imgPrimary;
            ExternalIDs = externalIDs;
        }
    }
}

File: src/app/models/resources/season.ts
import { Episode } from "./episode";
import {ExternalID} from "../external-id";
import {IResource} from "./resource";

export interface Season extends IResource
{
	seasonNumber: number;
	title: string;
	overview: string;
	episodes: Episode[];
	externalIDs: ExternalID[]
}
